src/android/InAppBrowser.java

```diff
@@ -1362,11 +1364,59 @@ public WebResourceResponse shouldInterceptRequest (final WebView view, String ur
         @TargetApi(Build.VERSION_CODES.LOLLIPOP)
         @Override
         public WebResourceResponse shouldInterceptRequest(WebView view, WebResourceRequest request) {
-            return shouldInterceptRequest(request.getUrl().toString(), super.shouldInterceptRequest(view, request), request.getMethod());
+            try {
+                Uri rawUrl = request.getUrl();
+                URL url = new URL(rawUrl.toString());
+                HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+                connection.setRequestMethod(request.getMethod());
+                for (Map.Entry<String, String> entry : request.getRequestHeaders().entrySet()) {
+                    connection.setRequestProperty(entry.getKey(), entry.getValue());
+                }
+                connection.connect();
+                boolean handlerSet = setCustomDownloadHandler(connection);
+                if(handlerSet) return null;
+                Map<String, List<String>> responseHeaders = connection.getHeaderFields();
+                Map<String, String> formattedResponseHeaders = new HashMap<String, String>();
+                for (Map.Entry<String, List<String>> entry : responseHeaders.entrySet()) {
+                    String name = entry.getKey();
+                    if (name == null)
+                        continue; // http/1.1 line
+                    formattedResponseHeaders.put(name, entry.getValue().toString());
+                }
+                return new WebResourceResponse(connection.getContentType(), connection.getContentEncoding(), connection.getResponseCode(), connection.getResponseMessage(), formattedResponseHeaders, connection.getInputStream());
+            }catch (IOException e) {
+                throw new RuntimeException(e);
+            }
         }
 
-        public WebResourceResponse shouldInterceptRequest(String url, WebResourceResponse response, String method){
-            return response;
+        /**
+         * Sets custom download handler.
+         *
+         * @param response
+         * @return boolean
+         */
+
+        private boolean setCustomDownloadHandler(HttpURLConnection response){
+            String[] downloadExt = {"zip", "binary", "tar", "rar"};
```

Comment:
I would think we want to look for the Content-Disposition header above all else.

Reply:
It specifically tells us that the url is a download, and optionally what the filename is.

Reply:
`Content-Disposition` is an optional header. We can be fully sure that it is a download, if we use content type.